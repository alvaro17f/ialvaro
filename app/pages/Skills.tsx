"use client";
import { domAnimation, LazyMotion, m, useScroll } from "framer-motion";
import data from "@/data/skills.json";
import Header from "@/components/Header";

export default function Skills() {
	return (
		<section id="skills">
			<Header title="Skills" />
			<div className="grid grid-cols-2 gap-5 md:grid-cols-4 place-items-center">
				{data.map(({ id, title, image, url }) => (
					<LazyMotion key={id} features={domAnimation}>
						<div className="grid place-content-center">
							<a href={url} rel="noreferrer" target="_blank">
								<m.div
									className="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
									initial={{ scale: 1, opacity: 0 }}
									whileInView={{ opacity: 1 }}
									viewport={{ once: false }}
									whileHover={{ scale: 1.1 }}
									whileTap={{ scale: 1 }}
									transition={{ duration: 0.7 }}
								>
									{/* <div className="h-32" /> */}
									<img src={image} alt={title} className="w-40 h-40 p-3" />
									<m.h1 key={title} className="mb-5 text-xl text-alvaro-white">
										{title}
									</m.h1>
								</m.div>
							</a>
						</div>
					</LazyMotion>
				))}
			</div>
		</section>
	);
}
