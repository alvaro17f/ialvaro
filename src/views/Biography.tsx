import { Content } from "src/components/Content";
import data from "src/data/biography.json";
import { type ChangeEvent, useEffect, useState } from "react";

export default function Biography() {
	const [isMobile, setIsMobile] = useState(true);
	const [slider, setSlider] = useState(Math.round((data.length - 1) / 2));
	const [selectedBio, setSelectedBio] = useState(data[slider]);

	useEffect(() => {
		setIsMobile(window.innerWidth < 768);
	});

	useEffect(() => {
		if (slider < data.length) {
			setSelectedBio(data[slider]);
		}
	}, [slider]);

	const handleSlider = (e: ChangeEvent<HTMLInputElement>) => {
		setSlider(Number(e.target.value));
	};

	return (
		<section
			id="biography"
			className="md:h-[100dvh] grid items-center pt-24 md:pt-0"
		>
			<Content>
				<div className="grid grid-cols-1 md:gap-10 md:grid-cols-[1fr_2fr]">
					<div className="mb-5">
						<a
							href="https://linkedin.com/in/alvarogarciamacias"
							rel="noopener noreferrer"
							target="_blank"
						>
							<img
								src="/images/profile/profile.png"
								alt="profile"
								width={250}
								height={250}
								className="border-[7px] rounded-full border-alvaro-white hover:border-dashed cursor-pointer"
							/>
						</a>
					</div>
					<div className="max-w-[75ch] md:h-[300px] items-center md:text-justify grid  whitespace-pre-wrap">
						<div>
							<h1 className="mb-5 text-4xl">
								Hello <span className="text-alvaro-danger">World</span>!
							</h1>
							<p>
								My name is{" "}
								<span className="text-xl text-alvaro-primary">√Ålvaro</span>, and
								I am a Full Stack Developer.
							</p>
							{selectedBio.bio}
						</div>
					</div>
				</div>
				<div className="grid w-full grid-cols-2 mt-20 md:grid-cols-3 place-items-center">
					<button
						type="button"
						onClick={() => slider !== 0 && setSlider((slider) => slider - 1)}
						className="p-2 border text-alvaro-primary rounded-xl border-alvaro-primary md:hover:bg-alvaro-primary md:hover:text-alvaro-base"
					>
						Shortest
					</button>
					{!isMobile && (
						<input
							id="minmax-range"
							type="range"
							min="0"
							max={data.length - 1}
							value={slider}
							onChange={handleSlider}
							className="w-full h-2 bg-gray-200 rounded-lg outline-none appearance-none cursor-pointer dark:bg-gray-700"
						/>
					)}
					<button
						type="button"
						onClick={() =>
							slider !== data.length - 1 && setSlider((slider) => slider + 1)
						}
						className="p-2 border text-alvaro-danger rounded-xl border-alvaro-danger md:hover:bg-alvaro-danger md:hover:text-alvaro-base"
					>
						Longest
					</button>
				</div>
			</Content>
		</section>
	);
}
