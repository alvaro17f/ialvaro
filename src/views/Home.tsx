import Wobble from "src/components/Wobble";

export default function Home() {
	return (
		<section
			id="home"
			className="h-[100dvh] top-0 grid place-items-center gap-5"
		>
			<div>
				<Wobble sentence="ALVARO" style="hover:text-alvaro-primary" />
				<Wobble sentence="GARCIA" style="hover:text-alvaro-danger" />
				<Wobble sentence="MACIAS" style="hover:text-alvaro-primary" />
			</div>
		</section>
	);
}
