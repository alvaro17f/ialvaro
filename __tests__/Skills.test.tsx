import { Skills } from "src/views/Skills";
import { cleanup, render, screen } from "@testing-library/react";
import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";

describe("<Skills />", () => {
	beforeEach(() => {
		const mockIntersectionObserver = vi.fn();
		mockIntersectionObserver.mockReturnValue({
			observe: () => null,
			unobserve: () => null,
			disconnect: () => null,
		});
		window.IntersectionObserver = mockIntersectionObserver;
		render(<Skills />);
	});
	afterEach(cleanup);

	it("should match the snapshot", () => {
		const { container } = render(<Skills />);
		expect(container).toMatchSnapshot();
	});

	it("should render a heading", () => {
		expect(
			screen.getByRole("heading", { level: 1, name: /skills/i }),
		).toBeDefined();
	});

	it("should render skills", () => {
		expect(screen.getAllByLabelText(/skill/i).length).toBe(12);
		expect(screen.getAllByRole("img").length).toBe(12);
	});
});
