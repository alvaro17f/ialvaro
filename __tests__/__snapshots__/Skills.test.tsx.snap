// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Skills /> > should match the snapshot 1`] = `
<div>
  <section
    id="skills"
  >
    <div
      class="h-24"
    />
    <section
      class="p-2 mb-5 rounded-lg text-alvaro-dark from-alvaro-primary to-transparent bg-gradient-to-r"
      style="width: 0%; opacity: 0; transform: scaleY(0.5) translateZ(0);"
    >
      <h1
        class="text-5xl md:text-3xl"
        style="opacity: 0;"
      >
        Skills
      </h1>
    </section>
    <div
      class="grid grid-cols-2 gap-5 md:grid-cols-4 place-items-center"
    >
      <div
        class="grid place-content-center"
      >
        <a
          href="https://react.dev/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="react.js"
              class="w-40 h-40 p-3"
              src="/skills/react.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              react.js
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://vuejs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="vue.js"
              class="w-40 h-40 p-3"
              src="/skills/vue.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              vue.js
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://typescriptlang.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="typescript"
              class="w-40 h-40 p-3"
              src="/skills/typescript.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              typescript
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://javascript.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="javascript"
              class="w-40 h-40 p-3"
              src="/skills/javascript.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              javascript
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://tailwindcss.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="tailwind"
              class="w-40 h-40 p-3"
              src="/skills/tailwind.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              tailwind
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://nextjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="next.js"
              class="w-40 h-40 p-3"
              src="/skills/next.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              next.js
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://rust-lang.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="rust"
              class="w-40 h-40 p-3"
              src="/skills/ferris.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              rust
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://go.dev/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="go"
              class="w-40 h-40 p-3"
              src="/skills/go.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              go
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://jestjs.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="jest"
              class="w-40 h-40 p-3"
              src="/skills/jest.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              jest
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://nodejs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="node.js"
              class="w-40 h-40 p-3"
              src="/skills/node.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              node.js
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://docker.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="docker"
              class="w-40 h-40 p-3"
              src="/skills/docker.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              docker
            </h2>
          </div>
        </a>
      </div>
      <div
        class="grid place-content-center"
      >
        <a
          href="https://mysql.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            aria-label="skill"
            class="grid gap-2 p-5 text-center border-dashed cursor-pointer place-items-center rounded-xl"
            style="opacity: 0; transform: none;"
            tabindex="0"
          >
            <img
              alt="sql"
              class="w-40 h-40 p-3"
              src="/skills/sql.svg"
            />
            <h2
              class="mb-5 text-xl text-alvaro-white"
            >
              sql
            </h2>
          </div>
        </a>
      </div>
    </div>
  </section>
</div>
`;
